app/src/main/java/com/a5corp/weather/GlobalActivity.java

```diff
@@ -35,15 +43,48 @@ protected void onResume() {
         }
 
         super.onResume();
-        Intent intent;
+        intent=new Intent();
 
         if (prefs.getLaunched()) {
-            intent = new Intent(GlobalActivity.this, WeatherActivity.class);
+            if(prefs.getCity()!=null && prefs.getCity().length()>0){
+                FetchWeather fw=new FetchWeather(this);
+                try {
+                    Info json=fw.execute(prefs.getCity()).get();
+                    intent = new Intent(GlobalActivity.this, WeatherActivity.class);
+                    intent.putExtra("json",json);
+                } catch (ExecutionException e) {
+                    e.printStackTrace();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }else{
+                FetchWeather fw=new FetchWeather(this);
+                try {
+                    float lat=prefs.getLatitude();
+                    float lon=prefs.getLongitude();
+                    Info json=fw.execute(String.valueOf(lat),String.valueOf(lon)).get();
+                    intent = new Intent(GlobalActivity.this, WeatherActivity.class);
+                    intent.putExtra("json",json);
+                } catch (ExecutionException e) {
+                    e.printStackTrace();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
```

Comment:
At StackOverFlow I found that we can pass objects through bundles or intent only if we make them implement serializable interface.

Reply:
Leave that part.

When `prefs.getLaunched()` is true, this means that 100% the name of the city is set. So why are you doing that if check in line 49? 
Moreover since the city name is already set, there is no need to do with the location, as it is to be used only when selecting the location `MenuItem` inside the `WeatherFragment`, so there is no need of that `else` part.

Reply:
Ok I will correct it.